File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_75477.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request


async def BenchmarkTest75477(request: Request):
    ua_value = request.headers.get('user-agent', '')
    data = '%s' % str(ua_value)
    size = min(int(data) if str(data).isdigit() else 0, 1024)
    data = bytearray(size)
    return {"updated": True}

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest75477.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest75477 {

    @PostMapping("/BenchmarkTest75477")
    public void BenchmarkTest75477(@RequestParam("field") String field, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String fieldValue = field != null ? field : "";
        String data = "" + fieldValue;
        String processed = org.owasp.encoder.Encode.forHtml(data);
        response.getWriter().print("<div>" + processed + "</div>");
    }
}
